### PR TITLE
Staging Sites: Disable SidebarNotice in the left menu

### PR DESCRIPTION
**Pull request by wojtekn**
Related to https://github.com/Automattic/dotcom-forge/issues/1937

## Proposed Changes

In this PR, I propose to hide the sidebar upsell component for staging sites.

|Before|After|
|---|---|
| Screenshot 2023-03-09 at 11 27 52 | Screenshot 2023-03-09 at 11 31 28 |

## Testing Instructions

1. Create staging site
2. Confirm that upsell is not displayed in the sidebar
3. Confirm that upsell is still displayed on the non-staging site

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by sejas — APPROVED** [review]
Code looks good.
I can see the Sidebar notice on normal business sites, and it's not visible on staging sites.

But I'm not able to see it on wpcalypso either.  Do I need to change something else?
I'm thinking that maybe we've deployed to production a fix removing the notices from the backend?

**Comment by wojtekn** [comment]
@sejas thanks for reviewing and testing it.

> But I'm not able to see it on wpcalypso either. Do I need to change something else?
I'm thinking that maybe we've deployed to production a fix removing the notices from the backend?

I see them on `wpcalypso` as well, although I noticed it is not displayed for me for some sites. The upsell content comes from the API request to the Jetpack site, and the JITM engine handles the message, so it's not related to the UI change we are making here.

File: client/my-sites/current-site/notice.jsx [hunk 1]
```diff
@@ -132,6 +134,7 @@ export default connect( ( state, ownProps ) => {
 		isDomainOnly: isDomainOnlySite( state, siteId ),
 		activeDiscount: getActiveDiscount( state ),
 		isSiteWPForTeams: isSiteWPForTeams( state, siteId ),
+		isStagingSite: ownProps.site.is_wpcom_staging_site,
```
Comment (danielbachhuber) [inline client/my-sites/current-site/notice.jsx]: Can we create an `isStagingSite` selector too, to follow the pattern immediately above?

Comment (sejas) [reply client/my-sites/current-site/notice.jsx]: I'm adding the selector here. https://github.com/Automattic/wp-calypso/pull/74216/commits/32c16eeaf632b813bdc88e856b28f40d9d22ab77

cc: @katinthehatsite

**Review by sejas — APPROVED** [review]
I've updated this branch and used the selector https://github.com/Automattic/wp-calypso/pull/74259/commits/6a91584240199b0d6d1528d2f71992737d3d42a7

File: client/my-sites/current-site/notice.jsx (hunk 1 above)
Comment (sejas) [reply client/my-sites/current-site/notice.jsx]: I've updated this branch to use the selector https://github.com/Automattic/wp-calypso/pull/74259/commits/6a91584240199b0d6d1528d2f71992737d3d42a7

Comment (wojtekn) [reply client/my-sites/current-site/notice.jsx]: Thanks for adding the selector and updating this branch!

**Pull request merged**